### DISCUSSION TITLE
Possible to pass metadata from the trigger onto ScaledJob?

### DISCUSSION THREAD

**Discussion by coolibre** [kedacore/keda#4431] [Q&A / Need Help]
We have a scenario where we leverage 1 ScaledJob per Redis List entry. We have a strict one pod per processing requirement due to a library used inside the application logic. Our operating users put items into the redis list per api. The ScaledJob pod starts, reads one element from the list and starts processing. Processing means it starts a subscription onto a broker and processes messages into a remote database. The redis entry is used to hand over metadata like the topic for the broker and other information.
The errors that might happen are reported via broker and reach the Ui that the operating user can restart the processing. When the user stops the process from the UI he puts a message into a redis publishing topic which is also listened by the application logic and the ScaledJob is exited with return code 0.

Goal: To remove the user from the picture where he is in direct charge of filling the list I need to perform automation from another service. That service should control the Keda trigger by filling the queue automatically and check the running processes for outages and restart the job. I cannot do this with returning exit codes only because the element with the metadata is already gone from the startlist!

Question: Is there a way with Keda to put metadata from the redis list onto the ScaledJob pods metadata to be able to fetch it from the controller via the kubernetes api. I need this to determine which processing might need a automated restart without having the user to monitor this.

Sorry if that is maybe not really related to Keda but maybe im missing something which could be useful.

**Comment by JorTurFer** [comment]
Hi,
Currently, KEDA doesn't have any kind of downward API
I don't get your requirement (most probably because I'm a noob with redis), what metadata would you like to share? `listName` for example?
